components/dev/site-info-debug.tsx

```
@@ -0,0 +1,131 @@
+
+/**
+ * Debug component that displays site information.
+ * Only use this in development environments.
+ */
+export function SiteInfoDebug() {
+  // Safeguard to prevent rendering in production
+  if (process.env.NODE_ENV === 'production') {
+    console.warn('SiteInfoDebug should not be used in production environments');
+    return null;
+  }
+  const { siteInfo, hasConsented, setConsent, isSyncing, lastSynced } = useSiteInfo();
```

ESLint

[error] React Hook "useSiteInfo" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)